File: server/app/schemas/agentic.py
from typing import Literal

from pydantic import BaseModel, Field


class SearchResult(BaseModel):
    title: str
    link: str
    snippet: str | None = None


class SearchContext(BaseModel):
    query: str
    scope: str
    result: SearchResult | None = None


class AgenticTurn(BaseModel):
    kind: Literal["question", "proposal"]
    content: str


class AgenticMessage(BaseModel):
    role: Literal["assistant", "user"]
    content: str


class AgenticState(BaseModel):
    status: Literal["idle", "question", "proposal", "accepted", "rejected"]
    turn: AgenticTurn | None = None
    proposal: str | None = None
    history: list[AgenticMessage] = Field(default_factory=list)
    search: SearchContext | None = None


class AgenticStartRequest(BaseModel):
    session_id: str


class AgenticRespondRequest(BaseModel):
    session_id: str
    answer: str


class AgenticDecisionRequest(BaseModel):
    session_id: str
    decision: Literal["yes", "no"]


class AgenticConversationResponse(BaseModel):
    agentic: AgenticState


class AgenticDecisionResponse(BaseModel):
    agentic: AgenticState

File: client/app/types/agentic.ts
export type AgenticSearchResult = {
  title: string;
  link: string;
  snippet?: string | null;
};

export type AgenticSearchContext = {
  query: string;
  scope: "city" | "prefecture";
  result?: AgenticSearchResult | null;
};

export type AgenticTurn = {
  kind: "question" | "proposal";
  content: string;
};

export type AgenticMessage = {
  role: "assistant" | "user";
  content: string;
};

export type AgenticState = {
  status: "idle" | "question" | "proposal" | "accepted" | "rejected";
  turn?: AgenticTurn | null;
  proposal?: string | null;
  history?: AgenticMessage[];
  search?: AgenticSearchContext | null;
};

export type AgenticConversationResponse = {
  agentic: AgenticState;
};

export type AgenticDecisionResponse = {
  agentic: AgenticState;
};
